### ISSUE TITLE
Using cotire for sped up builds

### ISSUE THREAD

**Issue by srajangarg** [symengine/symengine#1023]
Was trying out [cotire](https://github.com/sakra/cotire), branch is [here](https://github.com/srajangarg/symengine/tree/cotire).
Only compilation of library done, using GNU 5.3.1 on an i7-5600U
All times in seconds.
```
                       cotire   master   speedup

SymEngine   -j1        51.3     69.0     25.6%
			-j2        30.4     39.3	 22.6%

w Piranha   -j1        224.5    240.2    6.5%
            -j2        121.3    142.9    15.1%

w Flint     -j1        59.4     76.0     21.8%
            -j2        33.6     42.8     21.5%
```
I did not see considerable speedup in building/linking of executables.

**Comment by certik** [comment]
Nice!

We should add it to our cmake, as an option.

**Comment by isuruf** [comment]
Fixed in #1041
